native/vendor/google/billing/JniBilling.h

```diff
@@ -0,0 +1,83 @@
+#include "vendor/google/billing/result-values/BillingConfig.h"
+#include "vendor/google/billing/result-values/BillingResult.h"
+#include "vendor/google/billing/result-values/ExternalOfferReportingDetails.h"
+#include "vendor/google/billing/result-values/InAppMessageResult.h"
+#include "vendor/google/billing/result-values/Purchase.h"
+#include "vendor/google/billing/result-values/ProductDetails.h"
+#include "vendor/google/billing/result-values/UserChoiceDetails.h"
+#include "vendor/google/billing/build-params/BillingFlowParams.h"
+
+namespace cc {
+
+class CC_DLL JniBilling {
+public:
+    static BillingResult *toBillingResult(JNIEnv *env, jobject obj);
+    static std::vector<ProductDetails *> toProductDetailList(JNIEnv *env, jobject productsObj, jint startID);
+    static std::vector<Purchase *> toPurchaseList(JNIEnv *env, jobject productsObj, jint startID);
+    static BillingConfig *toBillingConfig(JNIEnv *env, jobject billingConfigObj);
+    static AlternativeBillingOnlyReportingDetails *toAlternativeBillingOnlyReportingDetails(JNIEnv *env, jobject alternativeBillingOnlyReportingDetailsObj);
+    static ExternalOfferReportingDetails *toExternalOfferReportingDetails(JNIEnv *env, jobject externalOfferReportingDetailsObj);
+    static InAppMessageResult *toInAppMessageResult(JNIEnv *env, jobject inAppMessageResultObj);
+    static ProductDetails *toProductDetail(JNIEnv *env, jobject productObj);
+    static Purchase *toPurchase(JNIEnv *env, jobject purchaseObj);
+    static ProductDetails::OneTimePurchaseOfferDetails *toOneTimePurchaseOfferDetails(JNIEnv *env, jobject obj);
+    static ProductDetails::InstallmentPlanDetails *toInstallmentPlanDetails(JNIEnv *env, jobject obj);
+    static ProductDetails::SubscriptionOfferDetails *toSubscriptionOfferDetails(JNIEnv *env, jobject obj);
+    static AccountIdentifiers *toAccountIdentifiers(JNIEnv *env, jobject obj);
+    static Purchase::PendingPurchaseUpdate *toPendingPurchaseUpdate(JNIEnv *env, jobject obj);
+    static ProductDetails::PricingPhase *toPricingPhase(JNIEnv *env, jobject obj);
+    static ProductDetails::PricingPhases *toPricingPhases(JNIEnv* env, jobject obj);
+    static BillingResult *callFunctionAndReturnBillingResult(const std::string &functionName, int tag, int callbackId);
+
+    static UserChoiceDetails::Product* toUserChoiceDetailsProduct(JNIEnv* env, jobject obj);
+    static UserChoiceDetails* toUserChoiceDetails(JNIEnv* env, jobject obj);
+
+
+    static jobject newBillingFlowParamsObject(int tag, BillingFlowParams* params);
+    static jobject newBillingClientBuilderObject(int tag, BillingClient::Builder* params);
+private:
+    static jobject newSubscriptionUpdateParamsObject(BillingFlowParams::SubscriptionUpdateParams* params);
+    static jobject newProductDetailsParamsObject(int tag, BillingFlowParams::ProductDetailsParams* params);
+    static jobject newProductDetailsParamsListObject(int tag, std::vector<BillingFlowParams::ProductDetailsParams*> listParams);
```

Comment:
Use const reference for object parameter.

Reply:
fixed.